src/lib/AudioUltra/Controls/Html5Player.ts

```diff
@@ -0,0 +1,82 @@
+import { WaveformAudio } from '../Media/WaveformAudio';
+import { Player } from './Player';
+
+export class Html5Player extends Player {
+  init(audio: WaveformAudio) {
+    super.init(audio);
+
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.on('resetSource', this.handleResetSource);
+
+    this.audio.el.addEventListener('play', this.handlePlayed);
+    this.audio.el.addEventListener('pause', this.handlePaused);
+  }
+
+  destroy() {
+    super.destroy();
+
+    if (this.audio?.el) {
+      this.audio.el.removeEventListener('play', this.handlePlayed);
+      this.audio.el.removeEventListener('pause', this.handlePaused);
+    }
+  }
+
+  protected playAudio(start?: number, duration?: number): void {
+    if (!this.audio || !this.audio.el) return;
+
+    this.audio.el.currentTime = this.currentTime;
+    this.audio.el.addEventListener('ended', this.handleEnded);
+    this.bufferPromise = new Promise(resolve => {
+      this.bufferResolve = resolve;
+    });
+
+    const time = this.currentTime;
+
+    this.audio.el
+      .play()
+      .then(() => this.bufferPromise!.then())
```

Comment:
could this be replaced with this more readable version?
```
Promise.all([
  this.audio.el.play(),
  this.bufferPromise,
])
```

Reply:
Yep, can replace that.